resources/sts/4.12/hypershift/InstallerPolicy.json

```diff
@@ -0,0 +1,74 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:CreateSubnet",
+                "ec2:CreateVolume",
+                "ec2:CreateVpc",
+                "ec2:DeleteSubnet",
+                "ec2:DeleteVolume",
+                "ec2:DeleteVpc",
+                "ec2:DescribeAddresses",
+                "ec2:DescribeAvailabilityZones",
+                "ec2:DescribeInternetGateways",
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DescribeRegions",
+                "ec2:DescribeReservedInstancesOfferings",
+                "ec2:DescribeRouteTables",
+                "ec2:DescribeSecurityGroups",
+                "ec2:DescribeSubnets",
+                "ec2:DescribeVpcAttribute",
+                "ec2:DescribeVpcs",
+                "ec2:DescribeVpcsWithContext",
+                "ec2:RunInstances",
+                "elasticloadbalancing:CreateLoadBalancer",
+                "elasticloadbalancing:DeleteLoadBalancer",
+                "iam:AddRoleToInstanceProfile",
+                "iam:AttachRolePolicy",
+                "iam:CreateInstanceProfile",
+                "iam:CreateOpenIDConnectProvider",
+                "iam:CreatePolicy",
+                "iam:CreatePolicyVersion",
+                "iam:CreateRole",
+                "iam:DeleteOpenIDConnectProvider",
+                "iam:DeletePolicyVersion",
+                "iam:DeleteRole",
+                "iam:DeleteRolePermissionsBoundary",
+                "iam:DetachRolePolicy",
+                "iam:GetOpenIDConnectProvider",
+                "iam:GetPolicy",
+                "iam:GetRole",
+                "iam:GetUser",
+                "iam:ListAttachedRolePolicies",
+                "iam:ListPolicyTags",
+                "iam:ListPolicyVersions",
+                "iam:ListRoleTags",
+                "iam:PutRolePermissionsBoundary",
+                "iam:TagPolicy",
+                "kms:DescribeKey",
+                "route53:ChangeResourceRecordSets",
+                "route53:DeleteDomainNsRecord",
+                "route53:DeleteHostedZone",
+                "route53:FindDomainNsRecord",
+                "route53:GetHostedZone",
+                "route53:ListHostedZones",
+                "route53:ListHostedZonesByName",
+                "route53:ListResourceRecordSets",
+                "route53:Route53",
+                "s3:CreateBucket",
+                "s3:DeleteBucket",
+                "s3:DeleteObject",
+                "s3:ListObjects",
```

Comment:
@karthikperu7 ListObjects is to satisfy this [call](https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/aws/client.go#L1240), I don't see any calls in cluster service listing S3 buckets do you?

Reply:
I got that from the IAM policy editor. The message was `Invalid Action: The action s3:ListObjects does not exist. Did you mean s3:ListBucket? The API called ListObjects authorizes against the IAM action s3:ListBucket.` [Learn more](https://docs.aws.amazon.com/IAM/latest/UserGuide/access-analyzer-reference-policy-checks.html#access-analyzer-reference-policy-checks-error-invalid-action)

Do you think it might be better to remove this action instead of replacing it with `ListBuckets` ?